share/config.py

```diff
@@ -264,3 +272,107 @@ def parse_config(config_yaml: str) -> Config:
         conf.add_input(current_input)
 
     return conf
+
+
+def aws_sm_expander(config_yaml: str) -> str:
+    needed_secret_arns = {}
+    needed_secret_keys = {}
+
+    re_pattern = r"arn:aws:secretsmanager:(?:[^:]*):(?:[^:]*):secret:(?:[^\n]*)"
+    found_secrets = re.findall(re_pattern, config_yaml)
+
+    for secret in found_secrets:
+        secret = secret.strip('"').strip("'")
+        splitted_secret = secret.split(":")
+
+        if len(splitted_secret) != 7 and len(splitted_secret) != 8:
+            raise SyntaxError("Invalid arn format: {}".format(secret))
+
+        if secret not in needed_secret_arns:
+            needed_secret_arns[secret] = ""
+
+        region = splitted_secret[3]
+        secrets_manager_name = splitted_secret[6]
+
+        if region == "":
+            raise ValueError("Must be provided region in arn: {}".format(secret))
+
+        if secrets_manager_name == "":
+            raise ValueError("Must be provided secrets manager name in arn: {}".format(secret))
+
+        if region not in needed_secret_keys:
+            needed_secret_keys[region] = {}
+
+        if secrets_manager_name not in needed_secret_keys[region]:
+            needed_secret_keys[region][secrets_manager_name] = ""
+
+    for region in needed_secret_keys:
+        for secrets_manager_name in needed_secret_keys[region]:
+            str_secrets = get_secret_values(secrets_manager_name, region)
+            parsed_secrets = parse_secrets_str(str_secrets)
+
+            needed_secret_keys[region][secrets_manager_name] = parsed_secrets
+
+    for secret in needed_secret_arns:
+        splitted_secret_string = secret.split(":")
+
+        region = splitted_secret_string[3]
+        secrets_manager_name = splitted_secret_string[6]
+
+        if len(splitted_secret_string) == 8:
+            wanted_key = splitted_secret_string[-1]
+
+            for got_key, got_value in needed_secret_keys[region][secrets_manager_name].items():
+
+                if got_key == wanted_key:
+                    needed_secret_arns[secret] = got_value
+
+        else:
+            needed_secret_arns[secret] = needed_secret_keys[region][secrets_manager_name]
+
+    for secret in needed_secret_arns:
+        if needed_secret_arns[secret] == "":
+            raise ValueError("Value for key: {} must not be empty".format(secret))
+
+        config_yaml = config_yaml.replace(secret, needed_secret_arns[secret])
+
+    return config_yaml
+
+
+def get_secret_values(secret_name: str, region_name: str) -> str:
+    session = boto3.session.Session()
+    client = session.client(service_name="secretsmanager", region_name=region_name)
+
+    try:
+        get_secret_value_response = client.get_secret_value(SecretId=secret_name)
+    except ClientError as e:
+        if e.response["Error"]["Code"] == "DecryptionFailureException":
```

Comment:
let's add some debug logging in all these if branch :)

Reply:
`shared_logger.debug("DecryptionFailureException: {}, secret: {}".format(e, secret_name))`  should we include the secret where it failed?

Reply:
yes, actually an `error` level would be better

prefer logging with payload as well, something like:
`shared_logger.error("get_secret_values", extra={"error_code": e.response["Error"]["Code"], "secret_name": secret_name})`